vernac/vernacentries.ml

```diff
@@ -2185,6 +2186,13 @@ let check_vernac_supports_polymorphism c p =
     | VernacExtend _ | VernacUniverse _ | VernacConstraint _) -> ()
   | Some _, _ -> user_err Pp.(str "This command does not support Polymorphism")
 
+(* Vernaculars that take a guarded flag *)
+let check_vernac_supports_guarded c p =
+  match p, c with
+  | None, _ -> ()
+  | Some _, (VernacFixpoint _ | VernacCoFixpoint _) -> ()
+  | Some _, _ -> user_err Pp.(str "This command does not support Guarded")
```

Comment:
@ejgallego Cf. here and the above function about the Polymorphic flag to better understand the attribute system.

Reply:
I am not sure what do you mean, I do understand this mechanism as I basically started it and IIRC I suggested / discussed the current design with Maxime some time ago. Code like this seems quite fragile, even if I am the one to blame and it is not even usable by vernacs declared using the interp table. It seems clear to me that we should not foster this wrapping style but indeed declaration-only attributes should be properly scoped.